src/common/snippets/include/snippets/op/tile_scheduler.hpp

```diff
@@ -0,0 +1,40 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "ngraph/op/op.hpp"
+#include "snippets/emitter.hpp"
+#include "tile.hpp"
+
+namespace ngraph {
+namespace snippets {
+namespace op {
+
+/**
+ * @interface TileScheduler
+ * @brief Contains a set of Tiles (currently one vector and one scalar) and performs necessary preparations
+ * before the Tiles could be executed: calculates offsets, sets proper work amounts, decrement pointers if the same data
+ * have to be read several times (broadcasting).
+ * @ingroup snippets
+ */
+class TileScheduler : public ngraph::op::Op {
```

Comment:
With "TileScheduler" appearance "Kernel" responsibilities are not clear at all. We need to more clearly state the architectural view on such decomposition, like:
- What's the kernel responsibility besides register allocation?
- Do you expect that Kernel will include several TileSchedulers?
- Do you expect that it might be required to have different types of TileScheduler (e.g. vector + scalar, matrix + scalar ...)? 
We can discuss offline if required.

Reply:
Discussed, moved offsets calculation to Kernel. Please also see [this comment](https://github.com/gdmitt/openvino/pull/3#issuecomment-1150916025)